### PR TITLE
Lock down test project firewall rule for dataprocCowTest

### PR DESCRIPTION
**Pull request by rogerwangcs**
Allowing the the default `allow-ingress-all` from all source IP's is insecure. This change restricts ingress traffic to only ip ranges required by dataproc.

https://broadinstitute.slack.com/archives/CADU7L0SZ/p1688063005145659

### CODE REVIEW

File: src/testFixtures/java/bio/terra/cloudres/google/compute/testing/NetworkUtils.java [hunk 1]
```diff
@@ -40,21 +40,27 @@ public static Network exceuteCreateNetwork(String projectId, boolean autoCreateS
...
+
     Firewall firewall =
         new Firewall()
             .setName(firewallName)
             .setNetwork(networkUrl)
             .setDirection("INGRESS")
-            .setAllowed(List.of(allowed));
+            .setSourceRanges(
+                List.of("10.128.0.0/9")) // Source range for internal dataproc VPC traffic
+            .setAllowed(List.of(new Allowed().setIPProtocol("all"))); // Allow all protocols
```
Comment (zloery) [inline src/testFixtures/java/bio/terra/cloudres/google/compute/testing/NetworkUtils.java:62]: Given that you're setting the `dataproc` tag on the cluster VMs, would `setSourceTags` also work for this firewall rule?

Comment (rogerwangcs) [reply src/testFixtures/java/bio/terra/cloudres/google/compute/testing/NetworkUtils.java:63]: Yeah! Added `setSourceTags` to the firewall.

**Review by zloery — APPROVED** [review]

File: src/testFixtures/java/bio/terra/cloudres/google/compute/testing/NetworkUtils.java [hunk 2]
```diff
@@ -40,21 +40,28 @@ public static Network exceuteCreateNetwork(String projectId, boolean autoCreateS
...
-    Allowed allowed = new Allowed().setIPProtocol("all");
+
     Firewall firewall =
         new Firewall()
             .setName(firewallName)
             .setNetwork(networkUrl)
             .setDirection("INGRESS")
-            .setAllowed(List.of(allowed));
+            .setSourceRanges(
+                List.of("10.128.0.0/9")) // Source range for internal dataproc VPC traffic
```
Comment (ddietterich) [inline src/testFixtures/java/bio/terra/cloudres/google/compute/testing/NetworkUtils.java:61]: Is that IP range always the case? I would have expected configuration.

**Review by ddietterich — APPROVED** [review]
I'm no firewall expert. Had one question.
The code looks fine AFAICT

File: src/testFixtures/java/bio/terra/cloudres/google/compute/testing/NetworkUtils.java (hunk 2 above)
Comment (rogerwangcs) [reply src/testFixtures/java/bio/terra/cloudres/google/compute/testing/NetworkUtils.java:61]: The [dataproc docs](https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/network#dataproc_connectivity_requirements) says that setting this range is sufficient. Some further digging shows this range is used by gcp ["Auto mode VPC networks"](https://cloud.google.com/vpc/docs/vpc#:~:text=Auto%20mode%20VPC%20networks%20are%20created%20with%20one%20subnet%20per,for%20future%20Google%20Cloud%20use.), which I think is what gcp networks are configured to by default.

**Pull request merged**